test/libyul/yulOptimizerTests/fullSuite/ackermann_function.yul

```diff
@@ -0,0 +1,559 @@
+// https://en.wikipedia.org/wiki/Ackermann_function
+// Test to see how FunctionSpecializer deals with functions that are too recursive / resource intensive.
+{
+    // 5
+    sstore(0, A(2, 1))
+    // 7
+    sstore(1, A(2, 2))
+
+    // Too many unrolling needed. In arbitrary precision, the value is 2**65536 - 3.
+    sstore(2, A(4, 2))
+
+    function A(m, n) -> ret {
```

Comment:
Can you create a second test with implementing the ackermann function using if+leave?

Like:
```
function A(m, n) -> ret {
  if eq(m, 0) {
    ret := add(n, 1)
    leave
  }

  if eq(n, 0) {
    ret := A(sub(m, 1), 1)
    leave
  }

  ret := A(sub(m, 1), A(m, sub(n, 1)))
}
```

Reply:
Added the test `ackermann_function_if.yul`